Python/suggestions.c

```diff
@@ -151,6 +151,9 @@ calculate_suggestions(PyObject *dir,
         if (item_str == NULL) {
             return NULL;
         }
+        if (PyUnicode_CompareWithASCIIString(name, item_str) == 0) {
```

Comment:
It can only be used if the second argument is ASCII string. `item_str` is UTF-8 string. The behavior in such case is not defined.

Reply:
Opened https://github.com/python/cpython/issues/106025